src/main/webapp/app/grading-system/base-grading-system/base-grading-system.component.ts

```diff
@@ -695,20 +695,12 @@ export abstract class BaseGradingSystemComponent implements OnInit {
 
     /**
      * Parse CSV file to a list of CsvGradeStep objects
+     * This method uses the 'papaparse' dependency, but Angular can not lazy load it in an abstract class.
+     * Therefore, the implementation is moved to the concrete classes (code duplication for the benefit of reducing main bundle size).
      * @param csvFile the read csv file
-     * @private
+     * @abstract
```

Comment:
Why did you duplicate the methods in the subclasses?

Reply:
> This method uses the 'papaparse' dependency, but Angular can not lazy load it in an abstract class. Therefore, the implementation is moved to the concrete classes (code duplication for the benefit of reducing main bundle size).